### PR TITLE
Upgrade to MapboxMaps v10.2

### LINKED ISSUE #3657: Upgrade to MapboxMaps v10.2.0
Now that [MapboxMaps v10.2.0](https://github.com/mapbox/mapbox-maps-ios/releases/tag/v10.2.0) is released, we should either relax the minimum MapboxMaps version requirement to be “from” v10.1.2 or increase it to v10.2.0 if we need anything that’s new to that release.

https://github.com/mapbox/mapbox-navigation-ios/blob/1ecc321139f3e08780aef7b6ed42f6e272ee2830/Package.swift#L28 https://github.com/mapbox/mapbox-navigation-ios/blob/1c6bbfa2f8e235a3a87ef18c31e9ff702a791884/Package.swift#L28

This is blocked until MapboxNavigationNative upgrades to MapboxCommon v21._x_.

/cc @mapbox/navigation-ios

### PR DESCRIPTION
**Pull request by jill-cardamon**
Fixes #3657.

### CODE REVIEW

**Comment by MaximAlien** [comment]
Please also update CocoaPods related files: `MapboxNavigation.podspec` and `Podfile` of CocoaPodsTest target.

File: Package.swift [hunk 1]
```diff
@@ -24,8 +24,8 @@ let package = Package(
     dependencies: [
         .package(name: "MapboxDirections", url: "https://github.com/mapbox/mapbox-directions-swift.git", from: "2.2.0-alpha.1"),
         .package(name: "MapboxMobileEvents", url: "https://github.com/mapbox/mapbox-events-ios.git", from: "1.0.0"),
-        .package(name: "MapboxNavigationNative", url: "https://github.com/mapbox/mapbox-navigation-native-ios.git", from: "81.0.0"),
-        .package(name: "MapboxMaps", url: "https://github.com/mapbox/mapbox-maps-ios.git", .exact("10.2.0-rc.1")),
+        .package(name: "MapboxNavigationNative", url: "https://github.com/mapbox/mapbox-navigation-native-ios.git", from: "82.0.1"),
+        .package(name: "MapboxMaps", url: "https://github.com/mapbox/mapbox-maps-ios.git", .exact("10.2.0")),
```
Comment (MaximAlien) [inline Package.swift:28]: I think we should not strictly depend on `v10.2.0` anymore, let's use `from: "10.2.0"`, as Navigation SDK v2.2.0 GA will use that version anyway.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,9 +4,9 @@
 
 ### Packaging
 
-* MapboxNavigation now requires [MapboxMaps v10.2.0-rc.1](https://github.com/mapbox/mapbox-maps-ios/releases/tag/v10.2.0-rc.1). ([#3614](https://github.com/mapbox/mapbox-navigation-ios/pull/3614))
+* MapboxNavigation now requires [MapboxMaps v10.2.0](https://github.com/mapbox/mapbox-maps-ios/releases/tag/v10.2.0-rc.1). ([#3665](https://github.com/mapbox/mapbox-navigation-ios/pull/3665))
```
Comment (MaximAlien) [inline CHANGELOG.md:7]:
```suggestion
* MapboxNavigation now requires [MapboxMaps v10._x_](https://github.com/mapbox/mapbox-maps-ios/releases/tag/v10.2.0). ([#3665](https://github.com/mapbox/mapbox-navigation-ios/pull/3665))
```

**Comment by jill-cardamon** [comment]
> Please also update CocoaPods related files: `MapboxNavigation.podspec` and `Podfile` of CocoaPodsTest target.

This is upgraded by update script right?

**Comment by MaximAlien** [comment]
> > Please also update CocoaPods related files: `MapboxNavigation.podspec` and `Podfile` of CocoaPodsTest target.
> 
> isn't this upgraded by update script?

I'm not aware of that change. I think we were always updating SPM and CocoaPods in sync manually (maybe worth automating this process).

File: MapboxCoreNavigation.podspec [hunk 3]
```diff
@@ -40,7 +40,7 @@ Pod::Spec.new do |s|
   s.requires_arc = true
   s.module_name = "MapboxCoreNavigation"
 
-  s.dependency "MapboxNavigationNative", "~> 81.0"
+  s.dependency "MapboxNavigationNative", "~> 82.0.1"
```
Comment (MaximAlien) [inline MapboxCoreNavigation.podspec:43]: Please set only minor version like:
```suggestion
  s.dependency "MapboxNavigationNative", "~> 82.0"
```

Comment (MaximAlien) [inline MapboxCoreNavigation.podspec:43]: Please set only minor version like:
```suggestion
  s.dependency "MapboxNavigationNative", "~> 82.0"
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,7 +4,7 @@
 
 ### Packaging
 
-* MapboxNavigation now requires [MapboxMaps v10.2.0](https://github.com/mapbox/mapbox-maps-ios/releases/tag/v10.2.0-rc.1). ([#3665](https://github.com/mapbox/mapbox-navigation-ios/pull/3665))
+* MapboxNavigation now requires [MapboxMaps v10._x_](https://github.com/mapbox/mapbox-maps-ios/releases/tag/v10.2.0-rc.1). ([#3665](https://github.com/mapbox/mapbox-navigation-ios/pull/3665))
```
Comment (MaximAlien) [inline CHANGELOG.md:7]:
```suggestion
* MapboxNavigation now requires [MapboxMaps v10._x_](https://github.com/mapbox/mapbox-maps-ios/releases/tag/v10.2.0). ([#3665](https://github.com/mapbox/mapbox-navigation-ios/pull/3665))
```

File: Package.swift [hunk 5]
```diff
@@ -25,7 +25,7 @@ let package = Package(
         .package(name: "MapboxDirections", url: "https://github.com/mapbox/mapbox-directions-swift.git", from: "2.2.0-alpha.1"),
         .package(name: "MapboxMobileEvents", url: "https://github.com/mapbox/mapbox-events-ios.git", from: "1.0.0"),
         .package(name: "MapboxNavigationNative", url: "https://github.com/mapbox/mapbox-navigation-native-ios.git", from: "82.0.1"),
```
Comment (MaximAlien) [inline Package.swift:27]:
```suggestion
        .package(name: "MapboxNavigationNative", url: "https://github.com/mapbox/mapbox-navigation-native-ios.git", from: "82.0.0"),
```

File: MapboxCoreNavigation.podspec (hunk 3 above)
Comment (MaximAlien) [reply MapboxCoreNavigation.podspec:43]: Please also run `$ pod update` in `Tests/CocoaPodsTest` because build job fails.

Comment (jill-cardamon) [reply MapboxCoreNavigation.podspec:43]: oh the build job failure fixed. I pushed prematurely.

File: MapboxCoreNavigation.podspec [hunk 6]
```diff
@@ -3,7 +3,7 @@ Pod::Spec.new do |s|
   # ―――  Spec Metadata  ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――― #
 
   s.name = "MapboxCoreNavigation"
-  s.version = '2.2.0-alpha.2'
+  s.version = '2.2.0-beta.1'
```
Comment (MaximAlien) [inline MapboxCoreNavigation.podspec:6]: It's better to use `$ ./scripts/update-version.sh VERSION_NUMBER`, as it'll apply these changes automatically. Other than that everything else looks good.

File: MapboxNavigation-SPM.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 7]
```diff
@@ -24,25 +24,25 @@
...
+          "version": "10.2.0"
         }
       },
       {
         "package": "MapboxDirections",
         "repositoryURL": "https://github.com/mapbox/mapbox-directions-swift.git",
         "state": {
           "branch": null,
-          "revision": "4d0afa5e6469a36084b9be48beb45706984826e0",
+          "revision": "28a0bc98ca13ac7cbfa08d06f273dcd94503d696",
```
Comment (MaximAlien) [inline MapboxNavigation-SPM.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:45]: Strange, I wonder why commit hash was changed, but version stayed the same...

**Review by MaximAlien — APPROVED** [review]

File: MapboxCoreNavigation.podspec (hunk 6 above)
Comment (jill-cardamon) [reply MapboxCoreNavigation.podspec]: i changed it back to do so using update script.

**Pull request merged**